### PR TITLE
MINIFICPP-1834 Add VERSIONINFO resource file

### PR DESCRIPTION
**Pull request by fgerlits**
https://issues.apache.org/jira/browse/MINIFICPP-1834

This adds version number, description, and some other info to the Windows binary.  To access this info, right-click on the executable file in a file explorer, select Properties and then the Details tab.  See sample screenshots in the Jira.

---

Thank you for submitting a contribution to Apache NiFi - MiNiFi C++.

In order to streamline the review of the contribution we ask you
to ensure the following steps have been taken:

### For all changes:
- [x] Is there a JIRA ticket associated with this PR? Is it referenced in the commit message?

- [x] Does your PR title start with MINIFICPP-XXXX where XXXX is the JIRA number you are trying to resolve? Pay particular attention to the hyphen "-" character.

- [x] Has your PR been rebased against the latest commit within the target branch (typically main)?

- [x] Is your initial contribution a single, squashed commit?

### For code changes:
- [ ] If adding new dependencies to the code, are these dependencies licensed in a way that is compatible for inclusion under [ASF 2.0](http://www.apache.org/legal/resolved.html#category-a)?
- [ ] If applicable, have you updated the LICENSE file?
- [ ] If applicable, have you updated the NOTICE file?

### For documentation related changes:
- [ ] Have you ensured that format looks appropriate for the output in which it is rendered?

### Note:
Please ensure that once the PR is submitted, you check GitHub Actions CI results for build issues and submit an update to your PR as soon as possible.

### CODE REVIEW

File: CMakeLists.txt [hunk 1]
```diff
@@ -20,7 +20,7 @@
 cmake_minimum_required(VERSION 3.17)
 cmake_policy(SET CMP0096 NEW) # policy to preserve the leading zeros in PROJECT_VERSION_{MAJOR,MINOR,PATCH,TWEAK}
 cmake_policy(SET CMP0065 OLD) # default export policy, required for self-dlopen
-project(nifi-minifi-cpp VERSION 0.12.0)
+project(nifi-minifi-cpp VERSION 0.13.0)
```
Comment (szaszm) [inline CMakeLists.txt:23]: There is another version number in libminifi/CMakeLists.txt

Comment (fgerlits) [reply CMakeLists.txt:23]: thanks -- fixed in 85fb3f6dbccb8588774173d2b80c832c07e27054

**Review by lordgamez — APPROVED** [review]

File: versioninfo.rc.in [hunk 2]
```diff
@@ -0,0 +1,44 @@
...
+FILEOS          VOS__WINDOWS32
+FILETYPE        VFT_APP
+BEGIN
+    BLOCK "StringFileInfo"
+    BEGIN
+        BLOCK "0409FDE9"
+        BEGIN
+            VALUE "LegalCopyright",   "Apache License v2.0"
+            VALUE "CompanyName",      "Apache Software Foundation"
+            VALUE "ProductName",      "MiNiFi C++"
```
Comment (szaszm) [inline versioninfo.rc.in:29]: I think we should make it possible to override the branding info as well from cmake, to make it easier to repackage and redistribute the software.

**Review by martinzink — APPROVED** [review]

File: versioninfo.rc.in (hunk 2 above)
Comment (fgerlits) [reply versioninfo.rc.in]: fixed in b2598b784de173952a58c0a8c79377bc08412b43 and 4c7a5c317b598b4a459f7ffc271785f00cbd3dc2

**Review by szaszm — APPROVED** [review]

**Review by adam-markovics — APPROVED** [review]

**Pull request closed without merging**